pulser-core/pulser/backend/noise_model.py

```diff
@@ -81,8 +81,11 @@ class NoiseModel:
             deviation of a normal distribution centered in 1.
         relaxation_rate: The rate of relaxation from the Rydberg to the
             ground state (in 1/µs). Corresponds to 1/T1.
-        dephasing_rate: The rate of a dephasing error occuring (in 1/µs).
-            Corresponds to 1/T2.
+        dephasing_rate: The rate of a dephasing occuring (in 1/µs) when a
+            Rydberg state is involved in the superposition. Corresponds
+            to 1/T2*.
+        hyperfine_dephasing_rate: The rate of dephasing occuring (in 1/µs)
+            between hyperfine ground states.
```

Comment:
What about harmonising the two definitions ?
```suggestion
        dephasing_rate: The rate of a dephasing occuring (in 1/µs) between
            Rydberg and ground states. Only used if Rydberg state is involved in the superposition. Corresponds
            to 1/T2*.
        hyperfine_dephasing_rate: The rate of dephasing occuring (in 1/µs)
            between hyperfine and ground states. Only used if hyperfine state is involved in the superposition.
```

Reply:
Well, `dephasing_rate` is also used in XY so saying it's between ground and Rydberg is not really accurate imo. I'll harmonise with a few changes